src/components/DataController/DataController.js

```diff
@@ -296,14 +328,90 @@ var DataController = createReactClass({
+      var makeModelSegmentor = function (selectedModelOutput, otherModelOutput) {
+        return function(dataseries) {
+          return dataseries[0].search(props.model_id) != -1 ? selectedModelOutput : otherModelOutput;
+        }
+      };
+      graph = assignColoursByGroup(graph, makeModelSegmentor(1, 0));
+      graph = fadeSeriesByRank(graph, makeModelSegmentor(1, .35));
+      graph = hideSeriesInLegend(graph, makeModelSegmentor(false, true));
+      graph = sortSeriesByRank(graph, makeModelSegmentor(1, 0));
+
+      //simplify graph by turning off tooltip and missing data gaps
+      graph.line.connectNull = true;
+      graph.tooltip = {show: false};
+      return graph;
+    };
+
+    //If the only thing changed since the last time the graph was rendered
+    //is the model, we already have all the data fetched, we just need to
+    //reformat the graph.
+    if(this.state.contextData &&
+        this.state.contextData.data.columns.length > 0 &&
+        _.isEqual(props.variable_id, this.props.variable_id) &&
+        _.isEqual(props.experiment, this.props.experiment)) {
+      var graph = emphasizeCurrentModel(this.state.contextData);
+
+      this.setState({
+        contextData: graph
+      });
+    }
+    else {
+      this.setContextGraphNoDataMessage("Loading Data");
+
+      var dataPromises = [];
+      var dataQueryParams = [];
+      var contextModels = _.without(_.uniq(_.pluck(props.contextMeta, "model_id")), props.model_id);
+      var params = _.pick(props, 'experiment', 'variable_id', 'ensemble_name');
+      params.timescale = 'yearly';
+      params.multi_year_mean = true;
+
+      params.model_id = props.model_id;
+      dataPromises.push(this.getDataPromise(params, "yearly", 0));
+      dataQueryParams.push(_.extend({}, params));
+
+      //determine which models have data that matches this variable and experiment
+      for(let i = 0; i < contextModels.length; i++ ) {
+        params.model_id = contextModels[i];
+        if(_.where(props.contextMeta, _.omit(params, 'ensemble_name')).length > 0){
+          dataPromises.push(this.getDataPromise(params, "yearly", 0));
+          dataQueryParams.push(_.extend({}, params));
+        }
+      }
+
+
+      //fetch data from the backend and assemble the graph
+      Promise.all(dataPromises).then(responses=> {
+        var data = _.pluck(responses, "data");
+        var graph = emphasizeCurrentModel(dataToLongTermAverageGraph(data, dataQueryParams));
+
+        this.setState({
+          contextData: graph
+        });
+      }).catch(error => {
+        this.displayError(error, this.setContextGraphNoDataMessage);
+      });
+    }
   },
 
   render: function () {
-    var longTermAverageData = this.state.longTermAverageData ? this.state.longTermAverageData : this.blankGraph;
-    var annualCycleData = this.state.annualCycleData ? this.state.annualCycleData : this.blankGraph;
     var statsData = this.state.statsData ? this.state.statsData : [];
```

Comment:
In the case that there is no `statsData`, this will return a new empty array each time, which will likely force unnecessary re-renders in whatever components are passed this state. Suggest you substitute a fixed empty array defined on `this` or even on the class (a static variable), as is done for the other data items.